src/librustdoc/html/markdown.rs

```diff
@@ -283,28 +285,36 @@ impl<'a, I: Iterator<Item = Event<'a>>> Iterator for CodeBlocks<'_, 'a, I> {
             ))
         });
 
-        let tooltip = if ignore != Ignore::None {
-            Some((None, "ignore"))
+        let (tooltip, special_class) = if ignore != Ignore::None {
+            (Some((None, "ignore")), " ignore")
         } else if compile_fail {
-            Some((None, "compile_fail"))
+            (Some((None, "compile_fail")), " compile_fail")
         } else if should_panic {
-            Some((None, "should_panic"))
+            (Some((None, "should_panic")), " should_panic")
         } else if explicit_edition {
-            Some((Some(edition), "edition"))
+            (Some((Some(edition), "edition")), " edition")
         } else {
-            None
+            (None, "")
         };
 
-        s.push_str(&highlight::render_with_highlighting(
-            text,
-            Some(&format!(
-                "rust-example-rendered{}",
-                if let Some((_, class)) = tooltip { format!(" {}", class) } else { String::new() }
-            )),
-            playground_button.as_deref(),
-            tooltip,
-            edition,
-        ));
+        if added_classes.is_empty() {
+            s.push_str(&highlight::render_with_highlighting(
+                text,
+                Some(&format!("rust-example-rendered{}", special_class)),
+                playground_button.as_deref(),
+                tooltip,
+                edition,
+            ));
+        } else {
+            let classes = if special_class.is_empty() {
+                added_classes.join(" ")
+            } else {
+                format!("{} {}", special_class.trim(), added_classes.join(" "))
+            };
+
+            s.push_str(&highlight::render_with_added_classes(text, classes, tooltip));
+        }
+
```

Comment:
Isn't this from that unrelated commit by Guillaume (cc https://github.com/rust-lang/rust/pull/79454/#issuecomment-751829416)?

Reply:
The removed parts are, yes, but the parts in green are from me (they notably add the support for the custom classes to the rendered html)